engine/src/mblandroidlcb.cpp

```diff
@@ -55,13 +55,13 @@ extern "C" JNIEXPORT void JNICALL Java_com_runrev_android_LCBInvocationHandler_d
 JNIEXPORT void JNICALL Java_com_runrev_android_LCBInvocationHandler_doNativeListenerCallback(JNIEnv *env, jobject object, jlong p_handler, jstring p_method_name, jobjectArray p_args)
 {
 #ifdef TARGET_SUBPLATFORM_ANDROID
-    extern bool MCAndroidIsOnSystemThread(void);
-    if (!MCAndroidIsOnSystemThread())
+    extern bool MCAndroidIsOnEngineThread(void);
+    if (!MCAndroidIsOnEngineThread())
     {
         typedef void (*co_yield_callback_t)(void *);
-        extern void co_yield_to_android_and_call(co_yield_callback_t callback, void *context);
+        extern void co_yield_to_engine_and_call(co_yield_callback_t callback, void *context);
         remote_call_t t_context = {p_handler, p_method_name, p_args};
-        co_yield_to_android_and_call(remote_call_func, &t_context);
```

Comment:
Isn't there a problem here with local JNI refs? The easiest way to fix this for now is to take GlobalRefs before doing the yield, and then delete them after the yield.